google/cloud/pubsub/message.h

```diff
@@ -40,6 +40,10 @@ pubsub::Message FromProto(::google::pubsub::v1::PubsubMessage);
 std::size_t MessageSize(pubsub::Message const&);
 // For Open Telemetry tracing only. Inserts or sets an attribute on the message.
 void SetAttribute(std::string const& key, std::string value, pubsub::Message&);
+// For Open Telemetry tracing only. Gets a pointer to the value of an given
+// attribute key on the message.
+char const* GetAttributeValue(std::string const& key, pubsub::Message& m);
```

Comment:
Generally we avoid returning unsafe types like a pointer.  Why not `std::string`?  You can use `.empty()` to represent "not present" or `absl::optional<std::string>` if that won't do.

Reply:
That's a much better thought.

I was implementing the Get function in PR#12667 which returns an opentelemetry::nostd::string_view, so I thought I needed the underlying ptr to the string value.

Now, I see the issue there was that it was returning a ptr to the temporary created map. A std::string object should stores the correct ptr so it should be fine.

Changed.